File: 942 - DI String Match.py
class Solution:
    def diStringMatch(self, s: str) -> list[int]:
        res = []
        l, r = 0, len(s)
        for c in s:
            if c == 'I':
                res.append(l)
                l += 1
            else:
                res.append(r)
                r -= 1
        return res + [l]

File: 942 - DI String Match.java
class Solution {
    public int[] diStringMatch(String s) {
        int l = 0, r = s.length();
        int[] res = new int[s.length() + 1];
        int i = 0;
        for (char c : s.toCharArray()) {
            if (c == 'I') {
                res[i++] = l++;
            } else {
                res[i++] = r--;
            }
        }
        res[i] = l;
        return res;
    }
}